source/docs/software/trajectory-end-to-end/entering-constants.rst

```diff
@@ -0,0 +1,102 @@
+
+.. tabs::
+
+  .. group-tab:: Java
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/ramsetecommand/Constants.java
+      :language: java
+      :lines: 44-54
+      :linenos:
+      :lineno-start: 44
+
+  .. group-tab:: C++ (Header)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/examples/RamseteCommand/include/Constants.h
+      :language: c++
+      :lines: 46-56
+      :linenos:
+      :lineno-start: 46
+
+DifferentialDriveKinematics
+---------------------------
+
+Additionally, we must create an instance of the ``DifferentialDriveKinematics`` class, which allows us to use the trackwidth (i.e. horizontal distance between the wheels) of the robot to convert from chassis speeds to wheel speeds.  As elsewhere, we keep our units in meters.
+
+.. tabs::
+
+  .. group-tab:: Java
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/ramsetecommand/Constants.java
+      :language: java
+      :lines: 32-34
+      :linenos:
+      :lineno-start: 32
+
+  .. group-tab:: C++ (Header)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/examples/RamseteCommand/include/Constants.h
+      :language: c++
+      :lines: 35-36
+      :linenos:
+      :lineno-start: 35
+
+Max Trajectory Velocity/Acceleration
+------------------------------------
+
+We must also decide on a nominal max acceleration and max velocity for the robot during path-following.  The maximum velocity value should be set somewhat below the nominal free-speed of the robod.  Due to the later use of the ``DifferentialDriveVolageConstraint``, the maximum acceleration value is not extremely crucial.
+
+.. tabs::
+
+  .. group-tab:: Java
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/ramsetecommand/Constants.java
+      :language: java
+      :lines: 62-63
+      :linenos:
+      :lineno-start: 62
+
+  .. group-tab:: C++ (Header)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/examples/RamseteCommand/include/Constants.h
+      :language: c++
+      :lines: 60-61
+      :linenos:
+      :lineno-start: 60
+
+Ramsete Parameters
+------------------
+
+Finally, we must include a pair of parameters for the RAMSETE controller.  The values shown below should work well for most robots, provided distances have been correctly measured in meters - for more information on tuning these values (if it is required), see :ref:`docs/software/wpilib-tools/path-planning/wpilib-trajectories/ramsete:Constructing the Ramsete Controller Object`.
```

Comment:
RamseteController has a default ctor for b = 2 and zeta = 0.7 btw.

Reply:
Probably best to keep it explicit anyway.